src/utils/DateUtils.ts

```diff
@@ -1,24 +1,125 @@
 import { Options } from '../misc/Options';
 import { Utils } from './Utils';
 import { l } from '../strings/Strings';
-import { TimeSpan } from './TimeSpanUtils';
-import * as Globalize from 'globalize';
 import * as _ from 'underscore';
-
+import * as moment from 'moment';
+/**
+ * The 'DateUtils' class renders dates and times using momentJS using the right culture languages and formats.
+ *
+ * This class can transform strings, numbers and date objects into a standardized ISO_8601 Date.
+ *
+ */
 export interface IDateToStringOptions {
+  /**
+   * Specifies whether the date and time are the current ones.
+   */
   now?: Date;
+
+  /**
+   * Specifies whether the date should be displayed normally or using Today, Yesterday and Tomorrow.
+   *
+   * **Examples**
+   *
+   * - If the date to render is equal to the current date, it will display 'Today':
+   *    Current date: march 8 2017, date to render: march 8 2017, result: 'Today'
+   *
+   * - If the date to render is equal to the current date plus or minus 1, it will display Yesterday or Tomorrow:
+   *    Current date: march 8 2017, date to render: march 7 2017, result: 'Yesterday'
+   *
+   * Default value is 'true'
+   */
   useTodayYesterdayAndTomorrow?: boolean;
+
```

Comment:
We should specify whether this option or the `useTodayYesterdayAndTomorrow` option takes precedence over the over. I'm guessing `useTodayYesterdayAndTomorrow` takes precedence, but this is a pretty wild guess. For instance, what happens if both options are `true` and I try to convert `Tuesday, March 9, 2017` with `now` being `Monday, March 8, 2017`? Do I get `Tomorrow` or `Next Tuesday`? What about a date that would result in `Today`?

Also, does it work both ways? For example, if current date is `Thursday, March 11, 2017` and date to render is 
`Monday, March 8, 2017`, would I get `Last Monday`?

We should definitely add the answers to these questions in other examples. For now, I suggest we go with something like this:

> Specifies whether to convert the Date object to the localized version of the corresponding day of the week, if the date to convert is part of the current week.
>
> **Example:**
> 
> If `useWeekdayIfThisWeek` is `true` and [`now`]{@link IDateToStringOptions.now} contains a Date object equivalent to `Monday, March 8, 2017`, then:
> - If the Date object to convert is equivalent to `Thursday, March 11, 2017`, the resulting string is the localized version of `Next Thursday`.
>
> Default value is `true`.